### PR TITLE
chore: improve controller-builder template and fix externalRef issues

### PR DESCRIPTION
**Pull request by yuwenma**
### Change description

Fixes #

### Tests you have done

- [ ] Run `make ready-pr` to ensure this PR is ready for review.
- [ ] Perform necessary E2E testing for changed resources.

### CODE REVIEW

File: dev/tools/controllerbuilder/template/controller/controller.go [hunk 1]
```diff
@@ -269,54 +259,70 @@ func (a *Adapter) Update(ctx context.Context, updateOp *directbase.UpdateOperati
...
 
 // Delete implements the Adapter interface.
 func (a *Adapter) Delete(ctx context.Context, deleteOp *directbase.DeleteOperation) (bool, error) {
 	log := klog.FromContext(ctx).WithName(ctrlName)
-	log.V(2).Info("deleting {{.Kind}}", "name", a.id.fullyQualifiedName())
+	log.V(2).Info("deleting {{.ProtoResource}}", "name", a.id.FullyQualifiedName())
 
-	if a.resourceID == "" {
-		return false, nil
-	}
```
Comment (yuwenma) [inline dev/tools/controllerbuilder/template/controller/controller.go:308]: This is actually the reason about this PR. Current controllers which use the old template shall fix it.

**Review by jingyih — APPROVED** [review]
/lgtm

File: dev/tools/controllerbuilder/template/controller/controller.go (hunk 1 above)
Comment (jingyih) [reply dev/tools/controllerbuilder/template/controller/controller.go:308]: I think this is OK? In the template, `a.resourceID` is set from either `spec.ResourceID` or `metadata.Name`.

Comment (yuwenma) [reply dev/tools/controllerbuilder/template/controller/controller.go:308]: You are right. I was fixing here and there and don't know where this comment comes from

**Pull request merged**